syntax = "proto3";

service TileServer {

  // Simple RPC
  // Obtains the tile at a given tile xyz
  rpc GetTile(TileXYZ) returns (Tile) {}
}

message TileXYZ {
  string tiff_path = 1;
  int32 x = 2;
  int32 y = 3;
  int32 z = 4;
}

message Tile {
  // Returns a tile as bytes
  bytes tile = 1;
}